pkg/vm/program.go

```diff
@@ -21,7 +32,19 @@ func DeserializeProgramJson(compiledProgram parser.CompiledJson) Program {
 		program.Data = append(program.Data, *memory.NewMaybeRelocatableFelt(felt))
 	}
 	program.Builtins = compiledProgram.Builtins
-	program.Identifiers = &compiledProgram.Identifiers
+	program.Identifiers = make(map[string]Identifier)
+	for key, identifier := range compiledProgram.Identifiers {
+		var programIdentifier Identifier
+		programIdentifier.FullName = identifier.FullName
+		programIdentifier.Members = identifier.Members
+		programIdentifier.Size = identifier.Size
+		programIdentifier.Decorators = identifier.Decorators
+		programIdentifier.PC = identifier.PC
+		programIdentifier.Type = identifier.Type
+		programIdentifier.CairoType = identifier.CairoType
+		programIdentifier.Value = lambdaworks.FeltFromDecString(identifier.Value.String())
+		program.Identifiers[key] = programIdentifier
+	}
```

Comment:
Could we use the same Identifier struct for the parser and the program?

Reply:
We use `	err = json.Unmarshal(byteValue, &cJson)` to unmarshall json into parser struct, but can't define a type felt there. It is possible but we would have to iterate through field to unmarshall them. This way we can point to reference while the structure is the same or add conversion functions when needed.